transformer_engine/pytorch/distributed.py

```diff
@@ -125,6 +131,35 @@ def gather_split_1d_tensor(
     return gathered
 
 
+@contextmanager
+def activation_recompute_forward(
```

Comment:
We should add this to the API docs, right? Since this is a new user-facing API.

Reply:
Or if it is hidden by another API - we should add that to the docs, since right now checkpointing support is not included there at all.

Reply:
Yes this isn't a user facing API. What is a part of the API is the `checkpoint` function in `distributed.py`. We could add that to the docs as a part of activation recompute support; NeMo does import and use it.